### ISSUE TITLE
Data type array is set to text in database

### ISSUE THREAD

**Issue by ndhoule** [balderdashy/sails-postgresql#36]
I recently noticed that whenever I create an attribute with type `array`, its type is set to `text` in Postgres. I'm using the Sails 0.10 branch and the Postgres adapter installed from the #associations branch. I had arrays working fine on 0.9.6, but switching to #associations (or Sails 0.10, hard to tell which) seems to have done it. I've also tried wiping out my `node_modules` folder and reinstalling just in case, no dice. No errors when starting up, etc. even in verbose mode.

Here's what my model looks like:

```javascript
// api/models/testModel.js
module.exports = {
  attributes: {
    // Doesn't work
    users: 'array',

    // Also doesn't work
    moreUsers: {
      type: 'array'
    }
  }
};
```

I popped open `psql` and did a describe on the table, and sure enough, the type is set to `text`.

**Comment by kevinburke** [comment]
Any update here? 

Postgres has pretty neat commands for working with JSON, but they generally expect objects at the top level, not arrays - they figure if you needed an array you would be using the default array type. Storing arrays as json means that you can't test for array membership directly in Postgres through Waterline, and you also can't by querying the database directly due to the lack of support - the only way is to pull up every item and do the check in application logic.

**Comment by zieglar** [comment]
Yes, I need this resolve too!

**Comment by sailsbot** [comment]
Thanks for posting, @ndhoule.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
